### PR TITLE
CI test for flavor-name syntax in scs-0103 yaml file

### LINKED ISSUE #339: CI tests for mandatory-flavor-list (#319)
With PR #319, we move away from the compressed form of listing mandatory and recommended SCS flavors to a more verbose format. While the compressed form only used SCS names and derived all properties from them, the new format lists all the properties, including the name as name and then also as extra_spec `scs:name-v1` and `scs:name-v2`.
https://github.com/SovereignCloudStack/standards/blob/issue/267/Tests/iaas/scs-0103-v1-flavors.yaml
This is more easy to read and also makes us less dependent on the names.
However, we have the same information twice now, once implicit in the name and once as explicit properties.
The risk is to have these diverge, creating inconsistencies that are embarassing for us and painful for operators and their customers. We need to avoid this.
TASKS:
- [ ] Use the existing code base to create a check for the consistency of scs-0103-v1-flavors.yaml file
- [ ] Run this code as a CI check, so we never merge a broken 0103-flavors file.

### PR DESCRIPTION
**Pull request by mbuechse**
closes #339

Test it like so:

```console
$ Tests/iaas/flavor-naming/check-yaml.py Tests/testing/
ERROR: flavor 'SCS-1V-4': field 'cpu-type' contradicting name-v2 'SCS-1V-4'; found 'crowded-core', expected 'shared-core'
ERROR: flavor 'SCS-2V-8': field 'name-v1' contradicting name-v2 'SCS-2V-8'; found 'SCS-2V-8', expected 'SCS-2V:8'
ERROR: flavor 'SCS-4V-16': field 'ram' contradicting name-v2 'SCS-4V-16'; found '12', expected '16.0'
ERROR: flavor 'SCS-1V-2': field 'cpus' contradicting name-v2 'SCS-1V-2'; found '2', expected '1'
ERROR: flavor 'SCS-2V-4-20s': field 'disk0-type' contradicting name-v2 'SCS-2V-4-20s'; found 'network', expected 'ssd'
ERROR: flavor 'SCS-4V-16-100s': field 'disk' contradicting name-v2 'SCS-4V-16-100s'; found '10', expected '100'
ERROR: file 'scs-0103-v1-flavors-wrong.yaml': found 6 errors
```

### CODE REVIEW

File: Tests/iaas/flavor-naming/check-yaml.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+CPUTYPE_KEY = {'L': 'crowded-core', 'V': 'shared-core', 'T': 'dedicated-thread', 'C': 'dedicated-core'}
+DISKTYPE_KEY = {'n': 'network', 'h': 'hdd', 's': 'ssd', 'p': 'nvme'}
+
+
+class Undefined:
+    def __repr__(self):
+        return 'undefined'
+
+
+class Checker:
```
Comment (artificial-intelligence) [inline Tests/iaas/flavor-naming/check-yaml.py:28]: opinion: can we add a docstring to this class?

File: Tests/iaas/flavor-naming/check-yaml.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+                    f"found {val!r}, expected {exp_val!r}"
+                )
+
+
+def main(argv):
+    if len(argv) != 2:
+        raise RuntimeError("must specify exactly one argument, PATH to the yaml file")
+    yaml_path = argv[1]
+    yaml_files = sorted([
+        fn
```
Comment (artificial-intelligence) [inline Tests/iaas/flavor-naming/check-yaml.py:75]: nit/opinion: I prefer longer, readable variable names, like `filename` in this case

**Review by artificial-intelligence — COMMENTED** [review]
LGTM beside some nits and opinions surrounding naming and documenting stuff.

File: Tests/iaas/flavor-naming/check-yaml.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env python3
+"""Check flavor names in the YAML files used for scs-0103-v1 with respect to scs-0100-vX.
```
[the remark this answers is not in the document]
Comment (mbuechse) [reply Tests/iaas/flavor-naming/check-yaml.py:2]: Why not. I will do that.

File: Tests/iaas/flavor-naming/check-yaml.py [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+Also check that the flavor properties encoded in the YAML match those encoded in the name.
+"""
+import importlib
+import os
+import os.path
+import sys
+
+import yaml
+
+fnmck = importlib.import_module("flavor-name-check")
```
[the remark this answers is not in the document]
Comment (mbuechse) [reply Tests/iaas/flavor-naming/check-yaml.py:13]: Well, I object to this whole importlib business altogether. I copied that line from some script by @garloff. But yes, of course, I will use a longer name.

File: Tests/iaas/flavor-naming/check-yaml.py (hunk 1 above)
Comment (mbuechse) [reply Tests/iaas/flavor-naming/check-yaml.py:28]: Yes, sure!

File: Tests/iaas/flavor-naming/check-yaml.py (hunk 2 above)
Comment (mbuechse) [reply Tests/iaas/flavor-naming/check-yaml.py:75]: I don't care either way, but I guess it would be better for some people, so I will use `filename`

**Review by artificial-intelligence — APPROVED** [review]
LGTM

**Pull request merged**